packages/@glimmer/manager/lib/internal/index.ts

```diff
@@ -213,13 +213,32 @@ export function getInternalComponentManager(
 ///////////
 
 export function hasInternalComponentManager(definition: object): boolean {
-  return getManager(COMPONENT_MANAGERS, definition) !== undefined;
+  return (
+    hasDefaultManager(COMPONENT_MANAGERS, definition) ||
+    getManager(COMPONENT_MANAGERS, definition) !== undefined
+  );
 }
 
 export function hasInternalHelperManager(definition: object): boolean {
-  return getManager(HELPER_MANAGERS, definition) !== undefined;
+  return (
+    hasDefaultManager(HELPER_MANAGERS, definition) ||
+    getManager(HELPER_MANAGERS, definition) !== undefined
+  );
 }
 
 export function hasInternalModifierManager(definition: object): boolean {
-  return getManager(MODIFIER_MANAGERS, definition) !== undefined;
+  return (
+    hasDefaultManager(MODIFIER_MANAGERS, definition) ||
+    getManager(MODIFIER_MANAGERS, definition) !== undefined
+  );
+}
+
+function hasDefaultManager(managerMap: WeakMap<object, object>, definition: object): boolean {
+  if (FEATURE_DEFAULT_HELPER_MANAGER) {
+    if (managerMap === HELPER_MANAGERS) {
```

Comment:
@chancancode I implemented it as you suggested but the map is only needed to determine the type of object we're dealing with. Since the utils that call this function already know the type, I think adding 3 dedicated helpers (ex. `hasDefaultHelperManager`) or moving the logic into the `has*` utils itself might be "better"?

Reply:
Yeah, splitting it into three sounds fine too, feel free to make the change 👍🏼 The other two can just be `return false` for now, but it will mark and obvious spot that we need to update when we implement something for the other two.

Having separate `hasDefault*Manager` functions shouldn't add much in terms of cost (probably will get inlined), and it seems clearer than doing everything in `hasInternal*Manager`.